src/source/Ice/SocketConnection.c

```diff
@@ -74,7 +74,11 @@ STATUS freeSocketConnection(PSocketConnection* ppSocketConnection)
         freeTlsSession(&pSocketConnection->pTlsSession);
     }
 
+#ifdef _WIN32
```

Comment:
Can I ask what's asserting here and why? 

If there is a strong reason to platform differentiate then I would recommend outsourcing the functionality to closeSocket() API which should be defined in Network.c/h where we already do conditional compilation based on platform/os

Reply:
Sorry if commit message is not clear enough.
For *nix system, `socket()` will create socket FD, so we can close it by `close()`.
For Windows, what `socket()` created is a not a FD, we can't use `close()` to a `SOCKET` object.

In [Win32 API Doc](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket)
> When a session has been completed, a closesocket must be performed.

Reply:
Thanks for clarification. We will take a note to hit this path with the tests. 

The changes themselves are good but I would recommend adding them to Network.c and have a common closeSocket similar to createSocket?

Reply:
I have updated PR with a common `closeSocket` in Network.c/.h.